### DISCUSSION TITLE
Serving WASM application with apache on the internet

### DISCUSSION THREAD

**Discussion by Caligino** [bevyengine/bevy#6007] [Q&A]
OK I used the wasm-setup and wasm-build scripts to compile my application on a remote http server instance. It all but manages to actually be accessible on the internet. on the website it shows a ```Forbidden``` access restriction where the module should load. I set up firewall-cmd to allow the port though and also specified to listen on port 4000 in httpd.conf.  I'm running out of relevant documentation to draw from but everything on the backend seems to be building correctly. I don't know if it's the system's internal configuration or apache's which is preventing it from loading. Has anyone else gotten this to work? If so, how? Thanks in advance.
<img width="828" alt="Screenshot 2022-09-17 191021" src="https://user-images.githubusercontent.com/8205312/190879213-af0eb9d3-4fdb-4944-966a-d55abcfd3729.png">
<img width="587" alt="Screenshot 2022-09-17 191103" src="https://user-images.githubusercontent.com/8205312/190879214-332bd2fd-54b5-416e-ab8b-529345027ff9.png">

**Comment by rparrett** [comment]
What `wasm-setup` and `wasm-build` scripts are you talking about? Is this from a third party game template?

There's no need to involve both Apache and `basic-http-server`. You can just copy the `index.html`, generated javascript and wasm files, and the assets to the web root directory.

**Reply by Caligino** [reply]
https://github.com/mfdorst/bevy-wasm
yes one step ahead of my self. i already did that and getting htaccess to work as intended is a pain in the a** as everyone knows. I get the forbidden message regardless of whether basic-html-server was running or not. i thought it was like an nginx alternative or something but further reading suggests it's only intended for development and learning purposes. So apache is capable but it doesnt seem willing.

**Reply by mockersf** [reply]
I would really not recommend serving something over the internet with apache httpd on your local machine unless you have a good understanding of what you're doing.

**Reply by rparrett** [reply]
Yeah, that's correct, it's just running a local server for your convenience while developing.

That's a pretty strange little repo, but if it created an `index.html` and you have it working locally, then it's just a matter of getting the files onto your remote webserver and configuring it to serve them.

I think this is likely to be some sort of file permissions issue on your remote server or some other issue with the apache configuration.

If you're not committed to running your own server, you could check out

https://github.com/NiklasEi/bevy_game_template which has a github workflow for deploying to a "github page." or
https://github.com/bevyengine/bevy_github_ci_template which is the "official bevy template" and has a github workflow for deploying to itch.io.

**Reply by Caligino** [reply]
> Yeah, that's correct, it's just running a local server for your convenience while developing.
> 
> That's a pretty strange little repo, but if it created an `index.html` and you have it working locally, then it's just a matter of getting the files onto your remote webserver and configuring it to serve them.
> 
> I think this is likely to be some sort of file permissions issue on your remote server or some other issue with the apache configuration.
> 
> If you're not committed to running your own server, you could check out
> 
> https://github.com/NiklasEi/bevy_game_template which has a github workflow for deploying to a "github page." or https://github.com/bevyengine/bevy_github_ci_template which is the "official bevy template" and has a github workflow for deploying to itch.io.

ok thanks yeah, i'm sure it has something to do with apache's or the host configuration

**Comment by mockersf** [comment]
if you want to reach your server on an external IP, you'll also need to listen to not 127.0.0.1 but to that IP (or 0.0.0.0 if you only have one)

**Comment by Caligino** [comment]
Thanks for everyone's input. i finally got it working properly. just have to polish up the presentation a little more. Just had to configure htaccess like i thought.
